android-sdk/src/main/java/org/sagebionetworks/bridge/android/AuthManagementActivity.java

```diff
@@ -0,0 +1,80 @@
+package org.sagebionetworks.bridge.android;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by liujoshua on 12/17/2017.
+ */
+
+public class AuthManagementActivity extends Activity {
+    private static final Logger logger =
+            LoggerFactory.getLogger(AuthManagementActivity.class);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        Uri data = intent.getData();
+        String study = data.getQueryParameter("study");
+        String email = data.getQueryParameter("email");
+        String token = data.getQueryParameter("token");
+
+        switch (data.getLastPathSegment()) {
+            case "startSession.html":
+                startSession(data);
+                break;
+            case "verifyEmail.html":
+                startSession(data);
+                break;
+            default:
+                startSession(data);
+        }
+
+        BridgeManagerProvider.getInstance()
```

Comment:
Seems like signInViaEmailLink may get called twice... here, and in startSession in the switch statement above, no?

Reply:
Yes, this code was moved to startSession but wasn't removed from here